platforms/cpuinfo.go

```diff
@@ -35,7 +34,7 @@ func cpuVariant() string {
 			var err error
 			cpuVariantValue, err = getCPUVariant()
 			if err != nil {
-				log.L.Errorf("Error getCPUVariant for OS %s: %v", runtime.GOOS, err)
+				panic(fmt.Sprintf("Error getCPUVariant for OS %s: %v", runtime.GOOS, err))
```

Comment:
panic seems too harsh here. Maybe just use stdlib `log`? (If we are really going to remove `containerd/log` here)

Reply:
If we are not able to reliably get the CPU variant, does containerd features like fetch, run etc work as expected?

Reply:
IMO It's really odd to use `log` package in one of `containerd` packages.
If we really want to isolate dependency, let's introduce a func or interface that can be overwritten.


<img width="636" alt="image" src="https://user-images.githubusercontent.com/865334/220729105-b8b372ed-e6f0-4f4e-abfc-44ee160bf2f7.png">

And have `platforms.Printf = log.L.Error` somewhere during initialization to preserve existing behavior.

Reply:
Should this initialization be made part of all the binaries which are in `cmd/` directory? Not able to get one location that serves all.

Reply:
What about changing this to return an error rather than print a log?

Reply:
@samuelkarp , changing the function to return an error also involves refactoring it in quite a number of places where the `cpuVariant()` function is used.  Thats why i was thinking, if the other solution suggested above of using the interface is much better.